deploy/ai-cloudformation.yaml

```diff
@@ -0,0 +1,299 @@
+        - Sid: DenyInsecureConnections
+          Effect: Deny
+          Principal: "*"
+          Action: s3:*
+          Resource: !Sub "arn:aws:s3:::${NuxeoS3Bucket}/*"
+          Condition:
+            Bool:
+              aws:SecureTransport: false
+
+  SageMakerS3Bucket:
+    Description: Store the data for SageMaker
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "ai-${Environment}-${AWS::Region}-sagemaker"
+      VersioningConfiguration:
+        Status: Enabled
+      Tags:
+        - Key: billing-category
+          Value: !Ref BillingCategory
+        - Key: billing-subcategory
+          Value: !Ref BillingSubCategory
+        - Key: project
+          Value: !Ref Project
+        - Key: contact
+          Value: !Ref Contact
+        - Key: environment
+          Value: !Ref Environment
+
+  SageMakerS3ArtifactBucketPolicy:
+    DependsOn: [SageMakerS3Bucket]
+    Description: Setting Amazon S3 bucket policy for AWS CodePipeline access
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref SageMakerS3Bucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Id: SSEAndSSLPolicy
+        Statement:
+        - Sid: DenyInsecureConnections
+          Effect: Deny
+          Principal: "*"
+          Action: s3:*
+          Resource: !Sub "arn:aws:s3:::${SageMakerS3Bucket}/*"
+          Condition:
+            Bool:
+              aws:SecureTransport: false
+
+  BackupS3Bucket:
+    Description: Store the MongoDB Backup of the Nuxeo base AI Service
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "ai-${Environment}-${AWS::Region}-mongodb-backup"
+      VersioningConfiguration:
+        Status: Enabled
+      Tags:
+        - Key: billing-category
+          Value: !Ref BillingCategory
+        - Key: billing-subcategory
+          Value: !Ref BillingSubCategory
+        - Key: project
+          Value: !Ref Project
+        - Key: contact
+          Value: !Ref Contact
+        - Key: environment
+          Value: !Ref Environment
+
+
+  BackupS3ArtifactBucketPolicy:
+    DependsOn: [BackupS3Bucket]
+    Description: Setting Amazon S3 bucket policy for AWS CodePipeline access
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref BackupS3Bucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Id: SSEAndSSLPolicy
+        Statement:
+        - Sid: DenyInsecureConnections
+          Effect: Deny
+          Principal: "*"
+          Action: s3:*
+          Resource: !Sub "arn:aws:s3:::${BackupS3Bucket}/*"
+          Condition:
+            Bool:
+              aws:SecureTransport: false
+
+  AIUser:
+    Type: AWS::IAM::User
+    Properties:
+      Path: "/"
+      UserName: !Sub "ai-${Environment}-${AWS::Region}"
+      ManagedPolicyArns:
+      - !Ref AIAccessBucketPolicy
+      - !Ref AIECRList
+      - arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
+
+  AIUserKeys:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      Serial: 1
```

Comment:
This is needed for key rotation. Incrementing this field will tell CF to rotate the key.